lib/ipmi_user.c

```diff
@@ -477,16 +481,47 @@ ipmi_user_build_password_prompt(uint8_t user_id)
  *
  * @returns pointer to char with password
  */
-char *
-ask_password(uint8_t user_id)
+static int
+ask_password(uint8_t user_id, char *buf, size_t buflen)
 {
-	const char *password_prompt =
-		ipmi_user_build_password_prompt(user_id);
-# ifdef HAVE_GETPASSPHRASE
-	return getpassphrase(password_prompt);
-# else
-	return (char*)getpass(password_prompt);
-# endif
+	const char *password_prompt;
+	char *sbuf;
+	size_t len;
+	int ret = -1;
+	char* (*password_routine)(const char *prompt);
+
+	/*
+	 * Note: getpass appears to be a trivial portable password routine,
+	 * however it is considered obsolete by most modern OSs. Unfortunately
+	 * there is not really a portable solution that was obvious so getpass
+	 * is currently to remain in use.
+	 */
+#ifdef HAVE_GETPASSPHRASE
+	password_routine = &getpassphrase;
+#else
+	password_routine = &getpass;
+#endif
```

Comment:
instead of a function pointer created with an ifdef, why not just a #define that sets it?

#ifdef HAVE_GETPASSPHRASE
#define password_routine getpassphrase
#else
#define password_routine getpass
#endif

Reply:
looks like this is moot if we are going with a termios/fgets situation anyway.